tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/gpu_backend_lib.cc

```diff
@@ -786,29 +786,82 @@ Status AMDGPUTargetModuleLinker(llvm::Module* module, GpuVersion gpu_version,
                                 const HloModuleConfig& hlo_module_config,
                                 const string& device_bitcode_dir_path) {
   // Link the input module with ROCDL.
-  auto amdgpu_version = absl::get_if<int>(&gpu_version);
+  auto amdgpu_version = absl::get_if<std::pair<int, std::string>>(&gpu_version);
   if (!amdgpu_version) {
     return xla::InternalError(
         "Incompatible AMD GCN ISA version was specified.");
   }
-  TF_RETURN_IF_ERROR(
-      LinkROCDLIfNecessary(module, *amdgpu_version, device_bitcode_dir_path));
+  TF_RETURN_IF_ERROR(LinkROCDLIfNecessary(module, amdgpu_version->first,
+                                          device_bitcode_dir_path));
 
   return Status::OK();
 }
 
-std::unique_ptr<llvm::TargetMachine> AMDGPUGetTargetMachine(
-    llvm::Triple target_triple, int amdgpu_version,
-    const HloModuleConfig& hlo_module_config) {
-  string feature_str = "+code-object-v3";
-#if TF_ROCM_VERSION >= 30900
-  // code-object-v3 is default, so no need to expliticitly specify it
-  // in the feature string. Also, starting with ROCm 4.0, this feature string
-  // is deprecated, and we get a warning to that effect. So removing that
-  // feature string
+// The following routine maps a feature token extracted from the
+// hipDeviceProp_t::gcnArchName string, and maps it to a valid feature_str
+// to be used for creating the AMDGPUTarget.
+// This mapping is currently in a state of flux because TF XLA uses its
+// own copy of LLVM, which is different from the LLVM version used by
+// hipcc/runtime in the ROCm install. Ordinarily this is not a problem,
+// but right now, the LLVM version used by hipcc/runtime has "targetID"
+// related changes which have not yet been upstreamed (to the LLVM repo)
+// When that upstreaming happens (and TF LLVM pointer moves past the
+// upstream commit), the following mapping will need to change
```

Comment:
Can this mapping check the compiler version and map it correctly? The new compiler version will use "+sramecc"/"-sramecc" instead of "+sram-ecc"/"-sram-ecc".

Reply:
This maybe possible in the future once the target-ID support has been upstreamed and we can associate that commit with a LLVM version number to check against. The essential problem is how to come up with a check that tells us whether or not the LLVM version (being used by the TF XLA code) has the changes related to target-ID support.